conda_smithy/ci_register.py

```diff
@@ -214,13 +214,19 @@ def travis_wait_until_synced(ignore=False):
     return content
 
 
-def travis_get_repo_info(user, project, show_error=False):
+def travis_get_repo(user, project, show_error=False, check_admin_access=True):
     headers = travis_headers()
     url = '{}/repo/{user}%2F{project}'.format(travis_endpoint, user=user, project=project)
     response = requests.get(url, headers=headers)
     try:
         response.raise_for_status()
         content = response.json()
+        if check_admin_access:
```

Comment:
I'd take all the return statements out of the ``try..except`` block at this point - the exception is getting too far away from the place where the exception may be getting raised.